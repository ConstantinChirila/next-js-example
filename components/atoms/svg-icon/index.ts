export * from './svg-icon'
export * from './svg-icon.styled'
export * from './svg-icon.types'
