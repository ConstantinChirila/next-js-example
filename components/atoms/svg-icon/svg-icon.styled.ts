import styled from 'styled-components'

export const StyledSVGIcon = styled.svg`
  display: block;
  height: 100%;
  width: 100%;
`
