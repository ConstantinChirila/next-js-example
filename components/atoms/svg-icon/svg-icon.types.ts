import { SVGAttributes } from 'react'

export type TSVGIconProperties = SVGAttributes<SVGSVGElement>
