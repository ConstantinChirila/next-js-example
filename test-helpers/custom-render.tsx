import { PropsWithChildren, ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { QueryClientProvider, QueryClient } from 'react-query'

import { defaultTheme } from '@tooploox-test/theme'
import { reactQueryDefaults } from '@tooploox-test/configuration'

function DefaultThemeProviderWithReactQuery({
  children,
}: PropsWithChildren<{}>): ReactElement {
  const queryClient = new QueryClient({
    ...reactQueryDefaults,
  })

  return (
    <ThemeProvider theme={defaultTheme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  )
}

function DefaultThemeProvider({
  children,
}: PropsWithChildren<{}>): ReactElement {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

function renderWithReactQueryContext(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProviderWithReactQuery, ...options })
}

function renderWithTheme(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProvider, ...options })
}

export { renderWithTheme, renderWithReactQueryContext }
