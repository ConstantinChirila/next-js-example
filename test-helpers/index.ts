export * from './custom-render'
export * from './setup'
