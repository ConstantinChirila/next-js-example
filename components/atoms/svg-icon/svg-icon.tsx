import React, { ReactElement } from 'react'

import { StyledSVGIcon } from './svg-icon.styled'
import { TSVGIconProperties } from './svg-icon.types'

export function SVGIcon(properties: TSVGIconProperties): ReactElement {
  return <StyledSVGIcon xmlns="http://www.w3.org/2000/svg" {...properties} />
}
