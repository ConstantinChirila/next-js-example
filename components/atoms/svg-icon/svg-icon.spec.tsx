import React from 'react'

import { SVGIcon } from './svg-icon'
import { renderWithTheme } from '../../../test-helpers'

describe('SVGIcon', () => {
  it('should match a snapshot', () => {
    const { container } = renderWithTheme(<SVGIcon>hello</SVGIcon>)
    expect(container).toMatchSnapshot()
  })
})
