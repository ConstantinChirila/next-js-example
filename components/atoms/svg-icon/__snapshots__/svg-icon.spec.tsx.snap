// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGIcon should match a snapshot 1`] = `
.c0 {
  display: block;
  height: 100%;
  width: 100%;
}

<div>
  <svg
    class="c0"
    xmlns="http://www.w3.org/2000/svg"
  >
    hello
  </svg>
</div>
`;
